lib/whitehall/document_collection_group_membership_updater.rb

```diff
@@ -0,0 +1,28 @@
+module Whitehall
+  class DocumentCollectionGroupMembershipUpdater
+    attr_reader :document
+
+    def initialize(document)
+      @document = document
+    end
+
+    def call
+      return unless document.published_edition
+
+      document_collection_group_memberships = DocumentCollectionGroupMembership.where(document_id: document.id)
+
+      return unless document_collection_group_memberships.any?
+
+      non_whitehall_link = DocumentCollectionNonWhitehallLink.create(
+        content_id: document.content_id,
+        title: document.published_edition.title,
+        base_path: Whitehall.url_maker.public_document_path(document.published_edition),
+        publishing_app: "content-publisher",
+      )
+
+      document_collection_group_memberships.each do |membership|
+        membership.update!(document_id: nil, non_whitehall_link_id: non_whitehall_link.id)
+      end
```

Comment:
Seems like a clever move to re-use the same non_whitehall_link model many times. Rather than doing a loop we can probably do an update_all?

Do you know if there are any risks with us re-using the same non_whitehall_link model. I don't think there is myself as I don't think there are any cases where we update the data in it, but wasn't sure if it gets auto deleted when a memebership is removed or anything.

Reply:
We talked about yesterday. It looks like nothing happens to the non whitehall link if the document collection group membership is deleted, but all associated document collection group memberships are deleted if non whitehall link is deleted.